httpclient5/src/main/java/org/apache/hc/client5/http/impl/auth/DigestScheme.java

```diff
@@ -488,4 +525,40 @@ public String toString() {
         return getName() + this.paramMap;
     }
 
+
+    /**
+     * Parses the provided array of characters and removes any surrounding quotes.
+     * This method is designed to process strings where the actual value is enclosed
+     * in quotes, and these quotes are not part of the value itself. It iterates
+     * through each character in the array, skipping the first and last quote characters,
+     * and appends the rest to a StringBuilder. The method is specifically tailored
+     * to handle cases where the entire string is enclosed in quotes.
+     *
+     * @param values The array of characters to be parsed. This array should
+     *               contain the characters of the string with potential surrounding
+     *               quotes.
+     * @return A String containing the characters from the input array with
+     * surrounding quotes removed. The method specifically skips the first
+     * and last quote characters if they exist, assuming that the entire
+     * string is enclosed in quotes.
+     */
+    private String parseAndRemoveQuotesAndDelimiters(final char[] values) {
```

Comment:
@arturobernalg I do not think this method is necessary. I believe `AuthChallengeParser` takes care of quote enclosed values. Please double-check.

Reply:
Absolutely right, @ok2c . I've verified and removed the unnecessary method. 

Thanks for pointing it out!